_docs/how-to/python-package.md

```diff
@@ -0,0 +1,207 @@
+where the individual modules would not obviously be separate Python packages
+-- in this case a single Python package will be easier to manage than multiple (perhaps interdependent) charmcraft libs.
+
+The main case where charmcraft libs are likely to be a good alternative to a Python package is for relations.
+In this case, the library is associated with a specific charm,
+it is likely to be simple enough for a single file (as a lot of logic likely lives in the related charms),
+and there is existing infrastructure and documentation supporting this pattern.
+However, if the relation library relies on any additional dependencies,
+it would probably still be better to make it a Python package.
+
+(python-package-name)=
+## Naming and namespacing your Python package
+
+For charm libraries intended for public use and distributed as Python packages,
+we recommend using the `charmlibs` namespace,
+particularly if your library is intended for other charmers to use.
+The package name should be `charmlibs-$libname`,
+imported as `from charmlibs import $libname`
+(or `import charmlibs.$libname as $libname` if you're stuck on an older `pyright` version that complains about the other form).
+
+The package should be a [namespace package](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/)
+using the `charmlibs` namespace.
+All this means is that your actual package is nested under an empty directory named `charmlibs`.
+Your file structure would look like `src/charmlibs/$libname/__init__.py`.
+There is no need to install the actual package named `charmlibs`.
+It exists on PyPI solely to reserve the package name as a namespace for charm libraries,
+and to aid charm library discoverability.
+
+If you have a dedicated repository for the charmlib, we recommend naming it `charmlibs-$libname` as well.
+For repositories containing several libraries, consider `$teamname-charmlibs`,
+with the individual packages following the naming and namespace recommendations.
+These naming  recommendations do not apply to other repository organisation schemes,
+but we still recomend the `charmlibs` namespace if your library is intended for public use.
+
+We don’t recommend using the `ops` or `charm` namespace for your charm libraries distributed as Python packages.
+It will be easier for charmers to follow your code if the `ops` namespace is reserved for the `ops` package.
+Likewise, the `charms` namespace is best left for charmcraft managed libs.
+
+If your library is only intended to be used by your own charms,
+for example if you write a library to be used by the machine and Kubernetes versions of your charm,
+releasing a package on PyPI using this naming scheme may not be the right approach,
+as you don't necessarily want to advertise the library for public consumption.
+
+(python-package-distribution)=
+## How to distribute your Python package
+
+Distributing your package on PyPI has the most benefits.
```

Comment:
Benefits for what or whom? It also has drawbacks (having to publish, claim the name, version, etc). I would skip this sentence and state the concrete trade-offs.